src/NHibernate/Mapping/ByCode/AbstractExplicitlyDeclaredModel.cs

```diff
@@ -16,6 +16,7 @@ public abstract class AbstractExplicitlyDeclaredModel : IModelExplicitDeclaratio
 		private readonly HashSet<MemberInfo> idBags = new HashSet<MemberInfo>();
 		private readonly HashSet<MemberInfo> lists = new HashSet<MemberInfo>();
 		private readonly HashSet<MemberInfo> manyToManyRelations = new HashSet<MemberInfo>();
+		private readonly HashSet<MemberInfo> itemManyToManyRelations = new HashSet<MemberInfo>();
```

Comment:
Now I'm confused. The many to many keys are gone. At the end you have to distinguish between many-to-many items and many-to-many keys. Just many-to-many and nothing more is not clear enough.